test/HydraStaking/VestedStaking.test.ts

```diff
@@ -325,5 +325,74 @@ export function RunVestedStakingTests(): void {
           .withArgs(this.staker.address, reward);
       });
     });
+
+    describe("calculatePositionTotalReward()", async function () {
+      it("should be bigger than claimable when maturing", async function () {
+        const { systemHydraChain, hydraStaking } = await loadFixture(this.fixtures.vestingRewardsFixture);
+
+        // add reward exactly before maturing (second to the last block)
+        const position = await hydraStaking.vestedStakingPositions(this.staker.address);
+        await time.setNextBlockTimestamp(position.end.sub(1).toNumber());
+        await commitEpochs(
+          systemHydraChain,
+          hydraStaking,
+          [this.signers.validators[0], this.signers.validators[1], this.staker],
+          1, // number of epochs to commit
+          this.epochSize
+        );
+
+        // calculate up to which epoch rewards are matured
+        const valRewardsHistoryRecords = await hydraStaking.getStakingRewardsHistoryValues(this.staker.address);
+        const valRewardHistoryRecordIndex = findProperRPSIndex(
+          valRewardsHistoryRecords,
+          position.end.sub(position.duration.div(2))
+        );
+
+        // enter maturing state
+        const nextTimestampMaturing = position.end.add(position.duration.div(2));
+        await time.setNextBlockTimestamp(nextTimestampMaturing.toNumber());
+        await mine();
+
+        const claimableRewards = await hydraStaking.calculatePositionClaimableReward(
+          this.staker.address,
+          valRewardHistoryRecordIndex
+        );
+        const totalRewards = await hydraStaking.calculatePositionTotalReward(this.staker.address);
```

Comment:
Should it be total created rewards or total rewards that are pending and claimable? 
Function NatSpecs are different from functionality.

Reply:
The later. The functionality is now fixed